src/includes/traits/Plugin/AutoCacheUtils.php

```diff
@@ -170,5 +170,49 @@ public function autoCacheCheckXmlSitemap($sitemap, $is_nested_sitemap = false, $
 
         return true;
     }
+
+    /**
+     * Download XML Sitemap via cURL and store in a temp file.
+     *
+     * @since 16xxxx
+     *
+     * @param string $url to XML Sitemap that needs to be downloaded
+     *
+     * @throws \Exception If unable to write temporary file or download XML Sitemap
+     *
+     * @return string Path to downloaded XML Sitemap file
+     */
+    public function autoCacheGetSitemapViaCurl($url)
+    {
+        if (!($tmp_dir = $this->getTmpDir())) {
+            throw new \Exception(__('No writable tmp directory.', SLUG_TD));
+        }
+
+        $tmp_file = $tmp_dir.'/'.$this->addTmpSuffix($tmp_dir).'.xml';
```

Comment:
Hmm, I see what you mean. `$tmp_dir.'/'.$this->addTmpSuffix($tmp_dir)`—I'm adding `$tmp_dir` to the path twice. 

I'll refactor that to use `wp_tempnam()` instead. Thanks!